### DISCUSSION TITLE
Call socket.io in middleware

### DISCUSSION THREAD

**Discussion by vitortoigo** [adonisjs/core#4265] [Help]
Hello guys, has a way to trigger socket.io in middleware directly?
Basically, i'm making a saas application and when a user login, needs to connect to the tenant room.

When i call the socket in middleware, returns undefined, but in controller works normally.

Congrats!

**Comment by MohamedAljoke** [comment]
hello it might be the order of initing the provider check ur adonis src json file for the provider of your websocket, if you could share a min project of your structure.
